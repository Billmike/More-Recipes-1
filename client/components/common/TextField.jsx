import React from 'react';
import PropTypes from 'prop-types';

const TextField = (props) => {
  const { id, name, type, defaultValue, placeholder, className,
    required, disabled, onChange } = props;

  return (
    <div className="row">
      <label htmlFor={name} className="col s3 black-text">
        {id}:
      </label>
      <div className="col s9">
        <input
          name={name}
          type={type}
          defaultValue={defaultValue}
          placeholder={placeholder}
          className={className}
          required={required}
          disabled={disabled}
          onChange={onChange}
        />
      </div>
    </div>
  );
};

TextField.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  defaultValue: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  className: PropTypes.string,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
};

TextField.defaultProps = {
  defaultValue: undefined,
  className: undefined,
  required: undefined,
  disabled: undefined,
};

export default TextField;
