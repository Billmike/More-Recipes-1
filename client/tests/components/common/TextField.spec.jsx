import React from 'react';

import TextField from '../../../components/common/TextField';

describe('TextField component', () => {
  const props = {
    id: 'title',
    name: 'title',
    type: 'text',
    defaultValue: null,
    placeholder: 'Title',
    className: 'className',
    required: false,
    disabled: false,
    onChange: jest.fn(),
  };
  it('should render component', () => {
    const wrapper = shallow(<TextField {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate change event', () => {
    const wrapper = shallow(<TextField {...props} />);

    wrapper.find('input').simulate('change');
    expect(props.onChange).toBeCalled();
  });
});
