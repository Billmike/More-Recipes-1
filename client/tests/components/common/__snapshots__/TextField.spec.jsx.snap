// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component should render component 1`] = `
<div
  className="row"
>
  <label
    className="col s3 black-text"
    htmlFor="title"
  >
    title
    :
  </label>
  <div
    className="col s9"
  >
    <input
      className="className"
      defaultValue={null}
      disabled={false}
      name="title"
      onChange={[MockFunction]}
      placeholder="Title"
      required={false}
      type="text"
    />
  </div>
</div>
`;
